### PR TITLE
ninjabackend: Use the right ranlib for static linker rules on macOS

### PR DESCRIPTION
**Pull request by nirbheek**
Fixes https://github.com/mesonbuild/meson/issues/12349

### CODE REVIEW

File: mesonbuild/backend/ninjabackend.py [hunk 1]
```diff
@@ -2355,7 +2355,12 @@ def generate_static_link_rules(self):
                 # ranlib, not to ar
                 cmdlist.extend(args)
                 args = []
-                cmdlist.extend(['&&', 'ranlib', '-c', '$out'])
+                # Ensure that we use the user-specified ranlib if any, and
+                # fallback to just picking up some ranlib otherwise
+                ranlib = self.environment.lookup_binary_entry('ranlib')
```
Comment (eli-schwartz) [inline mesonbuild/backend/ninjabackend.py:2360]: Doesn't this need the for_machine argument?

Comment (vanc) [reply mesonbuild/backend/ninjabackend.py:2360]: Ran into an error.

```
    ranlib = self.environment.lookup_binary_entry('ranlib')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Environment.lookup_binary_entry() missing 1 required positional argument: 'name'
```

Comment (nirbheek) [reply mesonbuild/backend/ninjabackend.py:2360]: Oops, I refactored and forgot that.

Comment (nirbheek) [reply mesonbuild/backend/ninjabackend.py]: @vanc please try again

**Comment by vanc** [comment]
Thanks a lot guys! I can confirm it now works without any issues.

**Review by dcbaker — APPROVED** [review]

**Pull request merged**